ui/app/contexts/metametrics.js

```diff
@@ -1,24 +1,92 @@
-import { Component } from 'react'
-import { connect } from 'react-redux'
+import React, { Component, createContext, useEffect, useRef, useCallback } from 'react'
+import { connect, useSelector } from 'react-redux'
 import PropTypes from 'prop-types'
-import { withRouter } from 'react-router-dom'
+import { withRouter, useLocation } from 'react-router-dom'
 import { compose } from 'redux'
 import {
   getCurrentNetworkId,
   getSelectedAsset,
   getAccountType,
   getNumberOfAccounts,
   getNumberOfTokens,
+} from '../selectors/selectors'
+import {
   txDataSelector,
-} from '../../../selectors'
-
-import { getEnvironmentType } from '../../../../../app/scripts/lib/util'
+} from '../selectors/confirm-transaction'
+import { getEnvironmentType } from '../../../app/scripts/lib/util'
 import {
   sendMetaMetricsEvent,
   sendCountIsTrackable,
-} from '../../utils/metametrics.util'
+} from '../helpers/utils/metametrics.util'
+
+export const MetaMetricsContext = createContext(() => {})
+
+export function MetaMetricsProvider ({ children }) {
+  const txData = useSelector(txDataSelector) || {}
+  const network = useSelector(getCurrentNetworkId)
+  const environmentType = getEnvironmentType()
+  const activeCurrency = useSelector(getSelectedAsset)
+  const accountType = useSelector(getAccountType)
+  const confirmTransactionOrigin = txData.origin
+  const metaMetricsId = useSelector((state) => state.metamask.metaMetricsId)
+  const participateInMetaMetrics = useSelector((state) => state.metamask.participateInMetaMetrics)
+  const metaMetricsSendCount = useSelector((state) => state.metamask.metaMetricsSendCount)
+  const numberOfTokens = useSelector(getNumberOfTokens)
+  const numberOfAccounts = useSelector(getNumberOfAccounts)
+  const currentPath = useLocation()
```

Comment:
useLocation probably isn't correct in retrospect as we need the full href and location from react-router is not the same object as the window API.